src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/services/GameInstanceService.kt

```
@@ -0,0 +1,81 @@
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Service
+
+@Service
+class GameInstanceService(
+  private val gameInstanceRepository: GameInstanceRepository,
+  private val gameService: GameService
+) : IService<GameInstance> {
+  override fun getAll(): List<GameInstance> {
+    return gameInstanceRepository.findAll()
+  }
+
+  override fun getSingle(id: String): GameInstance? {
+    return gameInstanceRepository.findOneById(ObjectId(id))
+  }
+
+  override fun create(request: Request): GameInstance {
+    val gameInstanceRequest = request as GameInstanceRequest
+    val igdbId = gameInstanceRequest.igdbId!! // TODO: GB-55 error handling
+    var game = gameService.getByIGDBId(igdbId)
+    if (game != null) {
+      val gameInstance = GameInstance(
+        igdbId = igdbId,
+        name = game.name,
+        platforms = game.platforms,
+        genres = game.genres,
+        universes = game.universes,
+        companies = game.companies,
+        releaseDate = game.releaseDate,
+        images = game.images
+      )
+      gameInstanceRepository.save(gameInstance)
+      return gameInstance
+    }
+    // if no game is found, call IGDB Client
+    game = gameService.create(GameRequest(igdbId))
+    val gameInstance = GameInstance(
+      igdbId = igdbId,
+      name = game.name,
+      platforms = game.platforms,
+      genres = game.genres,
+      universes = game.universes,
+      companies = game.companies,
+      releaseDate = game.releaseDate,
+      images = game.images
+    )
+    gameInstanceRepository.save(gameInstance)
+    return gameInstance
+  }
+
+  override fun delete(id: String) {
+    gameInstanceRepository.deleteById(id)
+  }
+
+  override fun update(model: GameInstance) {
+    gameInstanceRepository.save(model)
+  }
+
+  fun updateWithCustomFields(gameId: String, request: GameInstanceRequest) {
+    val gameInstance = gameInstanceRepository.findOneById(ObjectId(gameId))
```

Can defer this to GB-55 but we should return false/error out if the game is not found

---

gonna defer to GB-55 for simplicity sake